flash/core/data/data_source.py

```diff
@@ -352,6 +352,52 @@ def generate_dataset(
             return dataset
 
 
+class DataSourceCollection(dict):
```

Comment:
This is a decent idea but it adds complexity that doesn't need to be there.

DataModule already has these functions so why do they need to be here too?